### ISSUE TITLE
456

### ISSUE THREAD

**Issue by UnlcleTian** [UnlcleTian/kaggle#3]
import numpy as np
import pandas as pd
# from sklearn.model_selection import KFold
from lightgbm import LGBMClassifier

import plotly.offline as py
import plotly.graph_objs as go
import matplotlib.pyplot as plt
import seaborn as sns
import gc

# data_train = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\train.csv')

# data_test = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\test.csv')
# data_his_tra = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\historical_transactions.csv')

# data_mer = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\merchants.csv')

data_new_mer = pd.read_csv(r'C:\Users\guanhua.wang\Downloads\data\new_merchant_transactions.csv')

data_new_mer_dum1 = pd.get_dummies(data_new_mer.category_1,prefix='cate1',dummy_na=True)
data_new_mer_dum2 = pd.get_dummies(data_new_mer.category_3,prefix='cate3',dummy_na=True)

data_new_mer = pd.concat([data_new_mer,data_new_mer_dum1,data_new_mer_dum2],axis=1 ) \
                            .drop(['category_1','category_3'],axis=1)

print('new shape:',data_new_mer.shape)
# print(data_new_mer.columns)
# print(data_new_mer.columns.values)
print(data_new_mer.head())

data_new_mer['authorized_flag'] = data_new_mer['authorized_flag'].map({'Y':1,'N':0})

print (data_new_mer['authorized_flag'] )
agg_func={
     'authorized_flag':['sum','std'],
     'city_id':['nunique'],
     'installments':['mean','sum','min','max','std'],
     'cate3_A':['sum','nunique']
}
agg_new_mer= data_new_mer.groupby(['card_id']).agg(agg_func)

print(agg_new_mer.columns)
print(agg_new_mer.columns.values)

agg_new_mer.columns = ['_'.join(col).strip() for col in agg_new_mer.columns.values]
agg_new_mer.reset_index(inplace =True)

#print(agg_new_mer)
